### ISSUE TITLE
Repo Considerations

### ISSUE THREAD

**Issue by raganmd** [raganmd/TD-Examples#1]
Hi All, 

I'm excited to have a shared repo where we can exchange a lot of examples, but there are a few things we need to keep in mind. 

Please Consider:
- **Size** - When possible look carefully at the size of your commits. Code examples should not be very large when possible. There are some exceptions, of course, but in general these are about demonstrating techniques, not sharing full projects. Any given repo is limited to 1GB in size, and the more we keep an eye on this, the larger this example code set can be.
- **Assets** - In general assets **do not** belong in the repo. A jpeg here or there is okay, but large media files (anything over 15 MB) should never be added to the repo. It makes pulling difficult, and is takes up space that could be code instead. Consider sharing links via dropbox, or through another online medium for large movie files.
- **Organization** - Please keep your projects organized here. A shared repo is reflective of all there of us, so let's try and keep things tidy and organized.
- **cammelCase** - As a general rule I use cammelCase naming conventions, I think there's an argument to be made for underscores as well and I'm not particular about which one is used. What is important is that you're consistent in your practice.
- **Read Me Files** - We all need to be better about documentation, that said, please add read me files to your projects. If you add a read me to your directory it will display on the web, making it easy to build examples that are self documenting. Don't forget that your read me is for you, as much as it is for someone new to your project. I'm trying to get in the habit of using markdown as my read me method - you're welcome to make any choice you'd like, again just be consistent.
- **Commit Messages** - commit messages are hugely important, please remember to use them. Not only does it help you keep track of what you changed, it helps the rest of us in the repo know what changed, or why something changed. 
- **Duplicate and Version Files** - In general these do not belong here in the repo. They're fine to use as you're developing, but please delete any versioned files '_file.01.toe'_ before pushing. They're confusing and clutter up your directories. Again, if you'd like to keep them on your own machine, that's wonderful - just keep them out of the repo.

Glad to have a solid start to a shared repo here!

**Comment by barakooda** [comment]
No problem,
will do from next uploads.

if its too mess for now, then I or your can delete it.

**Comment by IanShelanskey** [comment]
I'll swap the 3D content over to Dropbox this weekend.

**Comment by lucesintetica** [comment]
Thanks a lot Mat, your research is a big richness for everyone of us. I'm
planning to start a teaching program for Every country in italy. Here se
have a very poor community of touchdesigner. I'd like to ask your point of
view about this topic to understand how is, in your opinion, the best
approach.

Thanks a lot for All..
Il 28/ago/2015 01:29, "Matthew Ragan" notifications@github.com ha scritto:

**Comment by raganmd** [comment]
Hi All - no worries about sizes, I just wanted to make sure we started in on some good practice earlier rather than later. I've learned quickly at Obscura that getting a solid handle on shared repositories is crucial to good code management, so it's better to have the conversation earlier rather than later. 

Lucas - shoot me an email if you'd like to talk about structuring courses. Ian just got back from teaching in Austin, so I'm sure he has some ideas as well.
